### DISCUSSION TITLE
FMOD Feature Wishlist

### DISCUSSION THREAD

**Discussion by bitbrain** [utopia-rise/fmod-gdextension#128] [Ideas]
Hello everyone!

I am currently thinking about the future of this addon and I thought it would be a great idea to ask for feedback from people actually using FMOD what they'd like to see in future.  I can start with my wish list of things:

- dedicated FMOD section in the project settings where I can specify which .bank files to load and where I can configure FMOD attributes within a UI (instead of code)
- preview FMOD event emitters in the editor itself (perhaps some sort of play checkbox?)
- https://github.com/utopia-rise/fmod-gdnative/issues/121 introduces `FmodEventEmitter2D` (previously called `FmodEventEmitter`) - introduce also `FmodEventEmitter3D` for 3D projects
- "auto discover" FMOD events and make them available within the editor itself. For example, creating an `FmodEventEmitter2D` could then offer a dropdown of all events contained by the .bank files (this is how it currently works in the Unity integration for FMOD)
- debug view in editor of event emitters. For example, we could draw debug lines to show the min/max radius of a sound effect or also its origin in the world

What are features you would expect to see from an FMOD integration that are not implemented yet?

**Comment by BSChad** [comment]
> "auto discover" FMOD events and make them available within the editor itself. For example, creating an FmodEventEmitter2D could then offer a dropdown of all events contained by the .bank files (this is how it currently works in the Unity integration for FMOD)

I know this is old and I saw your commit that adds the FmodGuidAndPath stuff to the Event Emitter. Before I touch any C++ code, is there a way to utilize the GuidAndPathPropertyEditor on a node from directly inside gdscript? For example, I have custom resources that store a sound to play and I'd love to have a picker instead of opening the explorer and copy pasting the path into a String property.
